huaweicloud/services/dms/resource_huaweicloud_dms_kafka_smart_connect.go

```diff
@@ -0,0 +1,329 @@
+			},
+			"node_count": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				ForceNew:    true,
+				Description: `Specifies the node count of the smart connect.`,
+			},
+		},
+	}
+}
+
+func resourceDmsKafkaSmartConnectCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	// createKafkaSmartConnect: create DMS kafka smart connect
+	var (
+		createKafkaSmartConnectHttpUrl = "v2/{project_id}/instances/{instance_id}/connector"
+		createKafkaSmartConnectProduct = "dmsv2"
+	)
+	createKafkaSmartConnectClient, err := cfg.NewServiceClient(createKafkaSmartConnectProduct, region)
+
+	if err != nil {
+		return diag.Errorf("error creating DMS Client: %s", err)
+	}
+
+	instanceID := d.Get("instance_id").(string)
+	createKafkaSmartConnectPath := createKafkaSmartConnectClient.Endpoint + createKafkaSmartConnectHttpUrl
+	createKafkaSmartConnectPath = strings.ReplaceAll(createKafkaSmartConnectPath, "{project_id}",
+		createKafkaSmartConnectClient.ProjectID)
+	createKafkaSmartConnectPath = strings.ReplaceAll(createKafkaSmartConnectPath, "{instance_id}", instanceID)
+
+	createKafkaSmartConnectOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	createKafkaSmartConnectOpt.JSONBody = utils.RemoveNil(buildCreateKafkaSmartConnectBodyParams(d))
+
+	// create kafka smart connect is allowd only when the instance status is RUNNING.
+	retryFunc := func() (interface{}, bool, error) {
+		createKafkaSmartConnectResp, createErr := createKafkaSmartConnectClient.Request("POST",
+			createKafkaSmartConnectPath, &createKafkaSmartConnectOpt)
+		retry, err := handleMultiOperationsError(createErr)
+		return createKafkaSmartConnectResp, retry, err
+	}
+	r, err := common.RetryContextWithWaitForState(&common.RetryContextWithWaitForStateParam{
+		Ctx:          ctx,
+		RetryFunc:    retryFunc,
+		WaitFunc:     KafkaInstanceStateRefreshFunc(createKafkaSmartConnectClient, instanceID),
+		WaitTarget:   []string{"RUNNING"},
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		DelayTimeout: 10 * time.Second,
+		PollInterval: 10 * time.Second,
+	})
+
+	if err != nil {
+		return diag.Errorf("error creating DMS kafka smart connect: %v", err)
+	}
+
+	kafkaSmartConnectRespBody, err := utils.FlattenResponse(r.(*http.Response))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	connectorId := utils.PathSearch("connector_id", kafkaSmartConnectRespBody, nil)
+	d.SetId(instanceID + "/" + connectorId.(string))
+
+	// enable smart connect, need to wait for the instance status to be RUNNING so that the job could be completed.
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"EXTENDING"},
+		Target:       []string{"RUNNING"},
+		Refresh:      dmsInstanceStatusRefreshFunc(region, cfg, d),
+		Timeout:      d.Timeout(schema.TimeoutCreate),
+		Delay:        60 * time.Second,
+		PollInterval: 10 * time.Second,
```

Comment:
```suggestion
		Delay:        5 * time.Second,
		PollInterval: 5 * time.Second,
```
and fix other places

Reply:
done